app/models/post.rb

```diff
@@ -1,5 +1,19 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
   has_many :comments
   has_many :likes
+
+  def update_post_counter
+    @author = User.find(author_id)
+    if @author.posts_counter
+      @author.posts_counter += 1
+    else
+      @author.posts_counter = 1
+    end
+    @author.save
+  end
```

Comment:
- [ ] **[OPTIONAL]** We already know the use who is interacting with the post object through `author` as defined in the association. So we don't really need to `find` it. We can also make this method more concise.

  ```
  def update_post_counter
    author.increment!(:posts_counter)
  end
  ```